Invenio3 plots

nikpap opened the pull request:
- [x] Thumbnails moved to left side.
- [x] Max height of text 
- [x] Mobile/responsiveness testing
- [x] Records with no plots show (1) plot
- [x] Implementation of javascript in different file
- [x] Display correct number of plots #
- [x] 1385669, 916908 are not working, whereas 782466, 922583 works fine
- [x] #slide-content-0 vs 1
- [x] Remove plot shifting left-right


* Desktop view
![capture d ecran 2016-02-25 a 14 14 06](https://cloud.githubusercontent.com/assets/2742143/13320547/13eaf452-dbca-11e5-90ba-bd2dfa49d529.png)

* Tablet view
![capture d ecran 2016-02-25 a 14 30 49](https://cloud.githubusercontent.com/assets/2742143/13320930/6d529476-dbcc-11e5-8728-4d8d36ce5072.png)

* Mobile view
![capture d ecran 2016-02-25 a 14 38 54](https://cloud.githubusercontent.com/assets/2742143/13321166/b4e3c9d0-dbcd-11e5-9419-081b1275e6af.png)


Signed-off-by: Nick Papoutsis <admin@nils.gr>

jmartinm commented on inspirehep/modules/theme/bundles.py:
```diff
@@ -44,7 +44,6 @@
         'js/inspire_base_init.js',
         filters='requirejs',
     ),
-    filters='jsmin',
```
Don't think this is meant to be commited :)

nikpap replied:
Nop, already changed it ;)

jmartinm commented:
Just note that the branch needs to be rebased with `invenio3` branch as there are some bundle changes there.

nikpap commented on inspirehep/modules/theme/templates/inspirehep_theme/format/record/Inspire_Default_HTML_detailed.tpl:
```diff
@@ -100,4 +113,43 @@
+    //     });
+
+    // });
+
```
@jmartinm I'm having some issues calling the `js/plots` file above. The error that appears is `Uncaught Error: undefined missing js/plots`.

Using the code inline (see below) works correctly. Adding `"js/plots"` in `inspire_base_init.js` also seems to work correctly.

Any suggestions?

jmartinm replied:
Yes, for a `require` call to work in the Jinja template it first needs to be part of a bundle (like adding it to `inspire_base_init.js`) You can check how we were doing before and we might need to create a JS bundle for the detail view?

jmartinm commented on inspirehep/modules/theme/bundles.py:
```diff
@@ -37,6 +37,7 @@
 
 js = NpmBundle(
     'js/inspire_base_init.js',
+    'js/detailed_record_init.js',
```
There should not be a need to add the file in both bundles.

jmartinm commented on inspirehep/modules/theme/static/js/detailed_record_init.js:
```diff
@@ -22,11 +22,11 @@
   "js/plots",
-  "js/citation_modal",
-  "js/datatables"
+  // "js/citation_modal",
```
Can you create another issue to make the citation modal and datatables work inside the detailed view?

nikpap replied:
Yes, issue created: https://github.com/inspirehep/inspire-next/issues/938.

jmartinm commented on inspirehep/modules/theme/static/js/plots.js:
```diff
@@ -25,29 +25,21 @@
     'jquery',
     'flight',
```
If you are not using `flight` you can remove this line

jmartinm commented on inspirehep/modules/theme/templates/inspirehep_theme/format/record/Inspire_Default_HTML_detailed.tpl:
```diff
@@ -100,4 +100,14 @@
+{% block javascript %}
+  {{ super() }}
+
+  {% assets %}
```
You would need the name of the asset here I would say:

```
{%- assets "inspirehep_author_update_js" %}
```

jmartinm commented on inspirehep/modules/theme/templates/inspirehep_theme/format/record/Inspire_HTML_detailed_macros.tpl:
```diff
@@ -265,27 +265,59 @@
 {% endmacro %}
 
 {% macro record_plots(record) %}
+  {% set vars = {'plotExists': False} %}
```
Same as in the other PR, if using the extension `do` simplifies the code I can add it before merging this PR.

nikpap replied:
Yes - that'd be actually great! I was wondering where this feature went, but eventually found a workaround. I'll refactor this code to work with `do`.

jmartinm commented on setup.py:
```diff
@@ -230,6 +230,7 @@ def run_tests(self):
             'almondjs = inspirehep.modules.theme.bundles:almondjs',
             'inspirehep_forms_css = inspirehep.modules.forms.bundles:css',
             'inspirehep_forms_js = inspirehep.modules.forms.bundles:js',
+            'detailed_plots_js = inspirehep.modules.theme.bundles:detailedjs',
```
we can name it `inspirehep_detailed_js`

jalavik commented:
:shipit: ?

nikpap commented:
@jalavik this PR is ready to be merged.

jmartinm commented:
I was trying it and I am getting the following error in the JS console.

```
Uncaught TypeError: Cannot read property 'offsetWidth' of undefined
```

And the plots do not work. Can you check?

nikpap commented:
@jmartinm There was indeed an issue with some plots not working correctly, I fixed it and rebased on latest invenio3. You can check records  1385669, 916908, 782466, 922583, they should be working correctly. Please let me know if `Uncaught TypeError: Cannot read property 'offsetWidth' of undefined` persists, the error doesn't appear on my side.

jmartinm commented:
Seems to work fine, no errors :)

Can you have a look at why every time the plot changes it shifts its position to the right at the end?

![plots](https://cloud.githubusercontent.com/assets/580727/13631994/c6d2ede6-e5e7-11e5-80a2-981650ee6649.gif)

nikpap commented:
@jmartinm updated, 'jerky' animation issue fixed. :+1:

jmartinm commented:
Yep, looks good. With your last changes, was the auto start of the carrousel removed?

nikpap commented:
Yeap. In an earlier discussion we had, I think we concluded that the carousel shouldn't automatically change pictures. Do you think I should re-add the option?

jmartinm commented:
Ok, I think thats fine for now. We can re-enable it in the future if needed.
